Update WND and Kerberos (#1209)
owncloud/docs-server: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc

--

.More WND Properties
Mounts to a Windows or Samba file server are labeled with a little four-pane Windows-style icon, and the left pane of your Files page includes a Windows Network Drive filter.

Files are synchronized bidirectionally, and you can create, upload and delete files and folders. ownCloud server admins can create Windows Network Drive mounts and optionally allow users to set up their own personal Windows Network Drive mounts.

> Depending on the authentication method, passwords for each mount are encrypted and stored in the ownCloud database, using a long random secret key stored in `config.php`. This allows ownCloud to access the shares when the users who own the mounts are not logged in. This access will not work if the mount is session based, where passwords are not stored and are available only for the current active session. In case another users are granted access to this mount, they will see a red triangle with an exclamation mark on the bottom right of the mount icon identifying lack of access.

Reviewer: ```suggestion
Depending on the authentication method, passwords for each mount are encrypted and stored in the ownCloud database, using a long random secret key stored in `config.php`. This allows ownCloud to access the shares when the users who own the mounts are not logged in. This access will not work if the mount is session based, where passwords are not stored and are available only for the current active session. In case other users are granted access to this mount, they will see a red triangle with an exclamation mark on the bottom right of the mount icon identifying lack of access.
```

> Depending on the authentication method, passwords for each mount are encrypted and stored in the ownCloud database, using a long random secret key stored in `config.php`. This allows ownCloud to access the shares when the users who own the mounts are not logged in. This access will not work if the mount is session based, where passwords are not stored and are available only for the current active session. In the case where another user is granted access to this mount, he will see a red triangle with and exclamation mark on the bottom right of the mount icon identifying lack of access.

Reviewer 2: ```suggestion
Depending on the authentication method, passwords for each mount are encrypted and stored in the ownCloud database, using a long random secret key stored in `config.php`. This allows ownCloud to access the shares when the users who own the mounts are not logged in. This access will not work if the mount is session based, where passwords are not stored and are available only for the current active session. In the case where another user is granted access to this mount, he will see a red triangle with an exclamation mark on the bottom right of the mount icon identifying lack of access.
```

=== Enabling External Storage

To enable external storage, as the ownCloud administrator go to menu:Settings[Admin > Storage] and tick the checkbox `Allow users to mount external storage`.

=== Creating a New Share

When you create a new WND share, you need multiple things::
+
--
* the server name or address hosting the share,
* the login credentials for the share if required by the authentication method,
* the share name and

> * optionally a sub-folder of the share you want to connect to.

Reviewer 2: ```suggestion
* optionally, a sub-folder of the share you want to connect to.
```

[CAUTION]
====
_Treat all the parameters as being case-sensitive._ Although some parts of the app might work properly regardless of casing, other parts might have problems if the case is not respected.
====

[IMPORTANT]
====
. When using *Kerberos* authentication, read the following documentation first:

.. xref:enterprise/authentication/kerberos.adoc[Kerberos Authentication]. +

> This documentation is a necessary prerequisite to setup Kerberos for the use with ownCloud.

Reviewer 2: ```suggestion
This documentation is a necessary prerequisite to setup Kerberos for use with ownCloud.
```

There are additional settings required when adding a mount point using the Kerberos authentication.

. When using *Credentials hardcoded in config file* authentication, read the following documentation first:

.. xref:enterprise/authentication/enterprise_only_auth.adoc#notes-for-credentials-hardcoded-in-config-file[Notes for Credentials Hardcoded in Config File]. +
This documentation is a necessary prerequisite to set up this authentication type.

.. xref:special-settings-when-using-credentials-hardcoded-in-config-file[Special Settings when using 'Credentials hardcoded in config file']. +
There are special settings required when adding a mount point using this authentication type.
====
--

Follow this procedure to create a new mount point based on WND::
+
--
. Enter the ownCloud mount point for your new WND share. This _must not_ be an existing folder.

. Select your authentication method. See xref:enterprise/external_storage/enterprise_only_auth.adoc[Enterprise-Only Authentication Options] for complete information on the available authentication methods.

. Enter the address of the server that contains the WND share.

. Use the share name provided by Windows or Samba.

> . Optionally, the root folder of the share. This is can be a subfolder name, or the `$user` variable when assigning the user's home directory. Note that the LDAP `Internal Username Attribute` must be set to the `samaccountname` for either the share or the root to work, and the user's home directory needs to match the `samaccountname`.

Reviewer: ```suggestion
. Optionally, the root folder of the share. This can be a subfolder name, or the `$user` variable when assigning the user's home directory. Note that the LDAP `Internal Username Attribute` must be set to the `samaccountname` for either the share or the root to work, and the user's home directory needs to match the `samaccountname`.
```

> . Optionally the root folder of the share. This is can be a subfolder name, or the `$user` variable when assigning the user's home directory. Note that the LDAP `Internal Username Attribute` must be set to the `samaccountname` for either the share or the root to work, and the user's home directory needs to match the `samaccountname`.

Reviewer 2: ```suggestion
. Optionally, the root folder of the share. This is can be a subfolder name, or the `$user` variable when assigning the user's home directory. Note that the LDAP `Internal Username Attribute` must be set to the `samaccountname` for either the share or the root to work, and the user's home directory needs to match the `samaccountname`.
```

(See xref:configuration/user/user_auth_ldap.adoc[User Authentication with LDAP].)

. Add the login credentials, if required, depending on the authentication method selected

. Sharing

.. Admin only menu:Settings[Admin > Storage]: +

> Admins can define that mount points can be shared to all (default), or restrct it to individuals or groups. Note that sharing is not available for all authorization methods. For details please see the

Reviewer 2: ```suggestion
Admins can define that mount points can be shared to all (default), or be restrictedt to individuals or groups. Note that sharing is not available for all authorization methods. For details please see the
```

> Admins can define that mount points can be shared to all (default), or be restrictedt to individuals or groups. Note that sharing is not available for all authorization methods. For details please see the

Reviewer: ```suggestion
Admins can define that mount points can be shared to all (default), or be restricted to individuals or groups. Note that sharing is not available for all authorization methods. For details please see the
```

> xref:enterprise/external_storage/enterprise_only_auth.adoc#authentication-option-details[Enterprise-Only Authentication Options]. In case, you will see the following on the moint point additionally: 

Reviewer 2: ```suggestion
xref:enterprise/external_storage/enterprise_only_auth.adoc#authentication-option-details[Enterprise-Only Authentication Options]. In this case, you will also see the following on the mount point: 
```

+
image::enterprise/external_storage/windows_network_drive/wnd-available-for.png[WND Sharing Options,width=400]

.. All menu:Settings[General > Storage]: +

> If sharing of mounts is allowed by the admin via menu:Settings[Admin > Storage > Allow sharing on user-mounted external storages], users can share the mount via menu:Files[Mountpoint > Details > Sharing]. To avoid accidentially sharing resources, the user must allow sharing a mount in the mount points view upfront by clicking on the gear icon as shown below. btn:[Enable sharing] appears only if sharing is generally allowed by the admin, see above.

Reviewer: ```suggestion
If sharing of mounts is allowed by the admin via menu:Settings[Admin > Storage > Allow sharing on user-mounted external storages], users can share the mount via menu:Files[Mountpoint > Details > Sharing]. To avoid accidentally sharing resources, the user must allow sharing a mount in the mount points view upfront by clicking on the gear icon as shown below. btn:[Enable sharing] appears only if sharing is generally allowed by the admin, see above.
```

> If sharing of mounts is allowed by the admin via menu:Settings[Admin > Storage > Allow sharing on user-mounted external storages], users can share the mount via menu:Files[Mountpoint > Details > Sharing]. To avoid accidentially sharing ressources, the user must allow sharing a mount in the mount points view upfront by clicking on the gear icon as shown below. btn:[Enable sharing] appears only if sharing is generally allowed by the admin, see above.

Reviewer: ```suggestion
If sharing of mounts is allowed by the admin via menu:Settings[Admin > Storage > Allow sharing on user-mounted external storages], users can share the mount via menu:Files[Mountpoint > Details > Sharing]. To avoid accidentially sharing resources, the user must allow sharing a mount in the mount points view upfront by clicking on the gear icon as shown below. btn:[Enable sharing] appears only if sharing is generally allowed by the admin, see above.
```

. Click the gear icon for additional mount options. Previews are enabled by default, but when using large storages with many files, you may want to disable previews as this can significantly increase performance.
+
image::enterprise/external_storage/windows_network_drive/wnd-gear-icon.png[WND Gear Icon,width=300]

. Your changes are saved automatically.

> . Finally, the mount point is created and will look like this as in the exampe below if a user has set it up:

Reviewer 2: ```suggestion
. Finally, the mount point is created and will look like this example if a user has set it up:
```

+
image::enterprise/external_storage/windows_network_drive/wnd-username-pwd.png[WND mount point created,width=500]

NOTE: When you create a new mountpoint using login credentials (session-based), you must log out of ownCloud and then log back in so you can access the share. You only have to do this the first time.
--

=== Special Settings when using 'Credentials hardcoded in config file'

The `Config key` to be entered as shown in the screen below, must be taken as described in  xref:enterprise/authentication/enterprise_only_auth.adoc#value-to-be-entered-in-the-mount-point-config-key-field[Value to be entered in the mount point config key field].

image::enterprise/external_storage/windows_network_drive/wnd-config-key.png[WND mountpoint and hardcoded credentials, width=500]

Note that this authentication method can only be used by ordinary users if the admin hands over the `Config key` created. This is secure, as no password is exposed.

> === Special Settings when Adding a Kerberos Mountpoint

Reviewer 3: ```suggestion
=== Special Settings when adding a Kerberos Mountpoint
```

Author: No, see https://headlinecapitalization.com 😄

> When the xref:enterprise/authentication/kerberos.adoc[Kerberos Authentication] has been setup, a necessary config key xref:configuration/server/config_apps_sample_php_parameters.adoc#a-map-of-servers-with-the-required-kerberos-data[wnd.kerberos.servers] needs to be provided upfront in `config.php`.

Reviewer 4: ```suggestion
When the xref:enterprise/authentication/kerberos.adoc[Kerberos Authentication] has been set up, a necessary config key xref:configuration/server/config_apps_sample_php_parameters.adoc#a-map-of-servers-with-the-required-kerberos-data[wnd.kerberos.servers] needs to be provided upfront in `config.php`.
```

* `Kerberos Server ID` +
This ID can be chosen freely and defines required parameters for the WND Kerberos setup. Multiple ID's with different setups can be created.
* `ockeytab` +
This is the keytab file that has been described in xref:enterprise/authentication/kerberos.adoc#keytab-files[Keytab Files].
* `ocservice` +

> This is the service user that has been described in xref:enterprise/authentication/kerberos.adoc#service-principal-name-spn[Service Principal Name (SPN)].

Reviewer 3: ```suggestion
This is the SPN of the service user that has been described in xref:enterprise/authentication/kerberos.adoc#service-principal-name-spn[Service Principal Name (SPN)].
```

Enter the `Kerberos Server ID` to define the server and Kerberos credentials as shown in the image below.

image::enterprise/external_storage/windows_network_drive/wnd-kerberos-1.png[WND mountpoint and Kerberos credentials, width=500]

> Note that this authentication method can only be used by ordinary users if the admin hands over the `Kerberos Server ID` created. Consider this as a very sensitive information like a password.

Reviewer 2: ```suggestion
Note that this authentication method can only be used by ordinary users if the admin hands over the `Kerberos Server ID` created. Consider this as very sensitive information like a password.
```

> Note that this authentication method can only be used by ordinary users if the admin hands over the `Kerberos Server ID` created. Consider this as very sensitive information like a password.

Reviewer 3: I would consider the `Keberos Server ID` as sensitive information. The keytab itself is, but the server id wouldn't help an attacker. The permission are still managed on the server and according the delegation rules.

Author: There are two sensitive infos:
* The keytab file itself (different story and covered in the Kerberos docs)
* The server ID not because of an attacker for hacking but to avoid giving access to shares
